packages/react-storage/src/components/StorageBrowser/views/LocationActionView/CopyView/useDestinationPicker.ts

```diff
@@ -0,0 +1,129 @@
+import { useCallback, useEffect, useRef } from 'react';
+import { isString } from '@aws-amplify/ui';
+import { usePaginate } from '../../hooks/usePaginate';
+import {
+  listLocationItemsHandler,
+  ListLocationItemsHandlerInput,
+  ListLocationItemsHandlerOutput,
+  LocationItemData,
+} from '../../../actions/handlers/listLocationItems';
+import { useGetActionInput } from '../../../providers/configuration';
+import { getDestinationListFullPrefix } from '../utils/getDestinationPickerDataTable';
+
+import { useDataState } from '@aws-amplify/ui-react-core';
+
+const DEFAULT_ERROR_MESSAGE = 'There was an error loading folders.';
+const DEFAULT_PAGE_SIZE = 1000;
+export const DEFAULT_LIST_OPTIONS = {
+  pageSize: DEFAULT_PAGE_SIZE,
+  delimiter: '/',
+};
+
+const DEFAULT_REFRESH_OPTIONS = { ...DEFAULT_LIST_OPTIONS, refresh: true };
+
+interface LocationActionInput extends ListLocationItemsHandlerInput {}
+
+interface LocationActionOutput {
+  items: LocationItemData[];
+  nextToken: string | undefined;
+}
+
+export const useDestinationPicker = ({
+  destinationList,
+}: {
+  destinationList: string[];
+}): {
+  items: ListLocationItemsHandlerOutput['items'];
+  hasNextToken: boolean;
+  currentPage: number;
+  isLoading: boolean;
+  hasError: boolean;
+  message: string | undefined;
+  handleNext: () => void;
+  handlePrevious: () => void;
+  range: [number, number];
```

Comment:
Can we just return the `pageItems` here rather than exposing `range` ?

Reply:
Will address in a follow up

Reply:
🍕🍕🍕🍕🍕🍕

Reply:
⭐ suck

Reply:
🐦

Reply:
You suck